src/lnpd/funding.rs

```diff
@@ -165,12 +165,22 @@ impl FundingWallet {
             fs::OpenOptions::new().read(true).write(true).create(false).open(wallet_path)?;
         let wallet_data = WalletData::strict_decode(&wallet_file)?;
 
-        let network = chain.try_into()?;
-        if DescriptorExt::<bitcoin::PublicKey>::network(&wallet_data.descriptor)? != network {
+        let target_network = chain.try_into()?;
+        let wallet_network = DescriptorExt::<bitcoin::PublicKey>::network(&wallet_data.descriptor)?;
+
+        let is_correct_network = match (wallet_network, target_network) {
+            (Network::Bitcoin, Network::Bitcoin) => true,
+            (Network::Testnet, _) => {
+                [Network::Testnet, Network::Regtest, Network::Signet].contains(&target_network)
+            }
```

Comment:
This can be done shorter:

```suggestion
            (Network::Testnet, Network::Testnet | Network::Regtest | Network::Signet) => true
```